external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -374,10 +377,22 @@ void outlineConvPartOps(Operation *convOp, ArrayRef<Operation *> secondOps,
   }
 }
 
+bool isConv2D(Operation *op) { return isa<tosa::Conv2DOp>(op); }
+
 // Inspired by / adapted from TestSCFIfUtilsPass in
 // test/lib/Transforms/TestSCFUtils.cpp.
 class TosaPartitionPass : public TosaPartitionBase<TosaPartitionPass> {
+  std::function<bool(Operation *)> isPrimaryOp;
```

Comment:
The command-line option specifies op names but the create-function takes a predicate;  both lead to setting isPrimaryOp.